rotkehlchen/globaldb/assets_management.py

```diff
@@ -48,53 +46,28 @@ def import_assets_from_file(
         raise InputError('The imported file is missing a valid list of assets')
 
     identifiers = []
-    for asset_data in data['assets']:
-        # Check if we already have the asset with that name and symbol. It is possible that
-        # we have added a missing asset. Using check_asset_exists for non ethereum tokens and
-        # for ethereum tokens comparing by identifier. The edge case of a non-ethereum token
-        # with same name and symbol will make this fail.
-        asset_type = asset_data['asset_type']
-        try:
-            if asset_type == AssetType.CUSTOM_ASSET:
-                custom_assets_db = DBCustomAssets(db_handler)
-                custom_assets_db.add_custom_asset(asset_data['extra_information'])
-                continue
-
-            asset_ref: Union[Optional[list[str]], Optional[AssetData]]
-            if asset_type == AssetType.EVM_TOKEN:
-                asset_ref = globaldb.get_asset_data(
-                    identifier=asset_data['identifier'],
-                    form_with_incomplete_data=True,
-                )
-            else:
-                asset_ref = globaldb.check_asset_exists(
-                    asset_type=asset_type,
-                    name=asset_data['name'],
-                    symbol=asset_data['symbol'],
-                )
-            if asset_ref is not None:
-                msg_aggregator.add_warning(
-                    f'Tried to import existing asset {asset_data["identifier"]} with '
-                    f'name {asset_data["name"]}',
-                )
-                continue
-
-            globaldb.add_asset(
-                asset_id=asset_data['identifier'],
-                asset_type=asset_type,
-                data=asset_data['extra_information'],
+    for asset in data['assets']:
+        with suppress(UnknownAsset):
+            # we expect that the asset is not in the database and this check raises UnknownAsset
+            asset.check_existence()
+            msg_aggregator.add_warning(
+                f'Tried to import existing asset {asset.identifier} with '
+                f'name {asset.name}',
```

Comment:
This is really hard to read code due to the way you use suppress.

Some questions:

1. Is it really important here to show a pop-up warning in the UI for each asset that already exists in the DB? What is the use case for this?
2. Clearer comment. And perhaps we need a boolean version of this. This code here is [using exceptions for control flow](https://softwareengineering.stackexchange.com/questions/189222/are-exceptions-as-control-flow-considered-a-serious-antipattern-if-so-why) in a not so straightforward way. I think a boolean function will be much better here.

```suggestion
        with suppress(UnknownAsset):
            # check if asset already exists in the DB and if yes skip importing it
            asset.check_existence(query_packaged_db=False)
            msg_aggregator.add_warning(
                f'Tried to import existing asset {asset.identifier} with '
                f'name {asset.name}',
```

Reply:
> Is it really important here to show a pop-up warning in the UI for each asset that already exists in the DB? What is the use case for this?

I think that normally all assets should be imported, and if they are not, it means a problem? For example a user might expect to see certain details of an asset after importing, but if asset with this identifier already exists, then currently existing details will be kept, and it would result in unexpected behaviour for the user.

Reply:
Also used a boolean method for this. Good idea